File: test_cv_update.py
#!/usr/bin/env python3
import sys
import os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), 'jobpilot-backend'))

from app import create_app
from app.models import User, CV

app = create_app()

with app.test_client() as client:
    with app.app_context():
        user = User.query.first()
        if not user:
            print("No test user found")
            sys.exit(1)
        
        cv = CV.query.filter_by(user_id=user.id).first()
        if not cv:
            print("No test CV found")
            sys.exit(1)
        
        cv_id = cv.id
        initial_years = cv.experience_years or 0
        
        print(f"Testing CV Update Feature")
        print(f"{'='*60}")
        print(f"CV ID: {cv_id}")
        print(f"Initial experience_years: {initial_years}")
        
        # Create a token - identity should be string
        from flask_jwt_extended import create_access_token
        token = create_access_token(identity=str(user.id))
        headers = {'Authorization': f'Bearer {token}'}
        
        new_years = 5
        print(f"Sending PUT request to update experience_years to {new_years}")
        put_response = client.put(
            f'/api/cv/update/{cv_id}',
            json={'experience_years': new_years},
            headers=headers
        )
        
        print(f"Status: {put_response.status_code}")
        data = put_response.get_json()
        print(f"Response: {data}")
        
        if put_response.status_code == 200:
            print("OK - PUT request successful!")
            from app import db
            db.session.expire_all()
            updated_cv = CV.query.get(cv_id)
            print(f"OK - Database: experience_years is now {updated_cv.experience_years}")
            if updated_cv.experience_years == new_years:
                print("OK - FEATURE WORKS: experience_years is editable and persists!")
            else:
                print(f"ERROR - Expected {new_years}, got {updated_cv.experience_years}")
        else:
            print(f"ERROR - PUT request failed with status {put_response.status_code}")

File: test_cv_update.js
#!/usr/bin/env node

const http = require('http');

function makeRequest(method, url, body, headers = {}) {
  return new Promise((resolve, reject) => {
    const urlObj = new URL(url);
    const options = {
      hostname: urlObj.hostname,
      port: urlObj.port || 80,
      path: urlObj.pathname + urlObj.search,
      method: method,
      headers: {
        'Content-Type': 'application/json',
        ...headers
      }
    };

    const req = http.request(options, (res) => {
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => {
        try {
          resolve({
            status: res.statusCode,
            body: JSON.parse(data)
          });
        } catch (e) {
          resolve({
            status: res.statusCode,
            body: data
          });
        }
      });
    });

    req.on('error', reject);
    if (body) req.write(JSON.stringify(body));
    req.end();
  });
}

async function testCVUpdate() {
  console.log('=== TEST: CV UPDATE API ===\n');

  try {
    // Login
    console.log('1. Logging in...');
    const loginRes = await makeRequest('POST', 'http://localhost:5000/api/auth/login', {
      email: 'test.features@jobpilot.com',
      password: 'TestPassword123!'
    });

    console.log('Login response:', JSON.stringify(loginRes.body, null, 2));
    const token = loginRes.body.access_token;

    if (!token) {
      console.error('❌ No token received');
      return;
    }

    // Get current CV
    console.log('\n2. Getting current CV...');
    const cvRes = await makeRequest('GET', 'http://localhost:5000/api/cv/current', null, {
      Authorization: `Bearer ${token}`
    });

    console.log('CV response status:', cvRes.status);
    console.log('CV response:', JSON.stringify(cvRes.body, null, 2));

    if (!cvRes.body.cv) {
      console.error('❌ No CV returned');
      return;
    }

    const cvId = cvRes.body.cv.id;
    console.log(`✓ CV ID: ${cvId}`);
    console.log(`✓ Current experience_years: ${cvRes.body.cv.experience_years}`);

    // Update CV
    console.log('\n3. Updating CV with experience_years = 1...');
    const updateRes = await makeRequest('PUT', `http://localhost:5000/api/cv/update/${cvId}`, {
      experience_years: 1
    }, {
      Authorization: `Bearer ${token}`
    });

    console.log('Update response status:', updateRes.status);
    console.log('Update response:', JSON.stringify(updateRes.body, null, 2));

    if (updateRes.status === 200) {
      console.log(`✓ Update successful!`);
      console.log(`✓ New experience_years: ${updateRes.body.cv?.experience_years}`);
    } else {
      console.error(`❌ Update failed with status ${updateRes.status}`);
    }

  } catch (error) {
    console.error('ERROR:', error.message);
  }
}

testCVUpdate();
